crates/re_arrow_store/src/store_gc.rs

```diff
@@ -71,6 +73,36 @@ pub struct Deleted {
     pub timeless: IntMap<EntityPathHash, IntMap<ComponentName, u64>>,
 }
 
+impl Deleted {
+    pub fn new(store_events: &[StoreEvent]) -> Self {
+        let mut this = Deleted {
+            row_ids: store_events.iter().map(|event| event.row_id).collect(),
+            timeful: Default::default(),
+            timeless: Default::default(),
+        };
+
+        for event in store_events {
+            if event.is_timeless() {
+                let per_component = this.timeless.entry(event.entity_path.hash()).or_default();
+                for component_name in event.cells.keys() {
+                    *per_component.entry(*component_name).or_default() +=
```

Comment:
shouldn't the delta be always negative? Or asked differently, isn't the right thing to do to first check whether the event is a deletion event? Seems strange not to do that

Reply:
Yes, this looks weird because this is in a temporary state. In future PRs this whole `Deleted` thing actually becomes `CompactedStoreEvents`, and works for both additions and deletions.